### ISSUE TITLE
Faceting on array fields

### ISSUE THREAD

**Issue by Jagdeep1** [marklogic-community/Corona#86]
In the current corona implementation array is not a range supported data type https://github.com/marklogic/Corona/wiki/Supported-Range-Datatypes  
I have a JSON document with structure like ... "aothor":"name", "published_date":"20-10-2011", "nouns":["n1","n2"], .... 

Here I want to do faceting on "nouns" field which has array data type.

**Comment by ryangrimm** [comment]
You are correct, at this time there is no support for adding a range index on a JSON array. However, I anticipate that the next major version of MarkLogic server will make supporting this rather trivial. In the meantime there are a couple options:

1) I could change the datamodel that's backing the JSON documents to support this (pretty painful for me)
2) You could use the support for XML strings inside JSON documents as a stopgap solution. Here's how I'd do that if I were you:

I'd create another key, "nouns::xml". Inside this key I'ld put a quoted string of XML that contains a <noun> element for every item in the array. So it would look something like:

{"author":"name", "published_date":"20-10-2011", "nouns":["n1","n2"], "nouns::xml":{"&lt;nouns&gt;&lt;noun&gt;n1&lt;/noun&gt;&lt;noun&gt;n2&lt;/noun&gt;&lt;/nouns>"}}

You'd then create an XML element range index on "noun" and query using that.

I know this isn't pretty and I'd like to have a better solution in place today but it is what it is.

**Comment by Jagdeep1** [comment]
Thanks.. I will try using till you come up with native support for array objects.
